rrule/src/core/rruleset.rs

```diff
@@ -128,8 +145,8 @@ impl RRuleSet {
     ///
     /// Limit must be set in order to prevent infinite loops.
     /// The max limit is `65535`. If you need more please use `into_iter` directly.
-    pub fn all(self, limit: u16) -> Result<Vec<DateTime>, RRuleError> {
```

Comment:
I think having the old `all` as a new function, like `some` makes sense.

- `some(50)` which returns just the first 50
- `all()` which returns all without a limit

My issue is that, it's not so visible in the API that you mostly need to set the limit.

Reply:
I would say we mostly need to use `set_limit`. Since I think `u16::MAX` still looks like unlimited to me. Even if we set the event to repeat every hour, it's _about_ events in a year!

🤔 I agree.

Maybe we can rename `all` to something more obvious and ugly :) like `all_with_no_limit`, `unlimited_all`, `dangerous_all` or `do_not_use_it_all` 🙈 😆

Reply:
> I would say we mostly need to use set_limit. Since I think u16::MAX still looks like unlimited to me.

That would be the ideal goal IMO of this new design to somehow make this limit transparent for most users while still protecting them. I think `u16::MAX` (i.e. 65_535) is a reasonable default that would make it very hard to find an RRuleSet that took more than 1 second of iteration before terminating, but also open to making the default something like `u16::MAX / 10` which I think would be okay too.

> Maybe we can rename all to something more obvious and ugly :) like all_with_no_limit, unlimited_all, dangerous_all or do_not_use_it_all 🙈 😆

Yeah, I was thinking `all_unchecked` or something like that could work. But my concern is that there would be quite a lot of methods on `RRuleSet`:
- `all(limit)` and `all_unchecked()`
- `all_with_error(limit)` and `all_with_error_unchecked()`
- + two variants for most of the current public methods
I don't know wether this is bad or not, but it somewhat annoys me a little bit 🤔

Reply:
I like `unchecked` but I also agree with your point that it almost doubles the size of the API and is ugly!
I'm still thinking. Generally I don't like what we have now, what you suggested and what I suggested :))

Reply:
I thought about it.

`RRuleSet` can have the following methods:
1. `.before(datetime)` (or `.to(datetime)` to avoid confusion with `.just_before(datetime)`)
2. `.after(datetime)` (or `.from(datetime)` to avoid confusion with `.just_after(datetime)`)
3. `.all()` to return all events between `from` and `to`
4. `.all_with_error()`, same as above but with errors
5. `.some(number)` same as above, but with limitation
6. `.some_with_error(number)`, same as above but with errors
7. `.just_after(datetime)`
8. `.just_before(datetime)`

Notes:
1. We can get rid of all other methods (like `between`s), since we can achieve all with the above methods
2. Since the current `all` method needs a `limit` parameter, people will get noticed about the change and most probably will read the documentation and/or CHANGELOG
3. We can name `all` and `some` to whatever you like :D

Reply:
After spending some time thinking. I indeed do like having a limit argument passed to the before, all, just_after etc methods, rather than my original design by hiding the `limit` value and having a default for it.

> We can get rid of all other methods (like betweens), since we can achieve all with the above methods

I don't quite see how you achieve the betweens with the mentioned methods. Do you mean that you can set the start date to a value and then call `.before(datetime)`?

> We have same amount of helper methods like now ;)

Wouldn't `.before` and `.after` need to have two versions each as well? One for unlimited and one with a limit

Reply:
> I don't quite see how you achieve the betweens with the mentioned methods. Do you mean that you can set the start date to a value and then call .before(datetime)?

Oh sorry, I think I didn't clarify the methods. In my suggestion, `.before(datetime)` and `.after(datetime)` would return `Self`.
So instead of calling `.all_between(start, end, ...)` we would call `.after(start).before(end).all()`.

Reply:
I love that. I think we are converging towards the new validation limits api :D I will try to have the implementation ready tomorrow.